lazyload/__init__.py

```diff
@@ -1,21 +1,50 @@
 import sys
+from types import ModuleType
 
-class LazyModule(object):
-    """Lazily import a module with minimal side effects"""
-    def __init__(self, module_name):
-        self.module_name = module_name
-        self.module = None
 
-    def __getattr__(self, attr):
-        if not self.module:
-            del sys.modules[self.module_name]
-            self.module = __import__(self.module_name)
+class _LazyModuleMarker(object):
+    """
+    A marker to indicate a LazyModule type.
+    Allows us to check module's with `isinstance(mod, _LazyModuleMarker)`
+    to know if the module is lazy.
+    """
+    pass
 
-            # future imports will get the real module
-            sys.modules[self.module_name] = __import__(self.module_name)
 
-        return getattr(self.module, attr)
+def make_lazy(module_path):
+    """
+    Mark that this module should not be imported until an
+    attribute is needed off of it.
+    """
+    sys_modules = sys.modules  # cache in the locals
```

Comment:
OK, was wondering if there was some other reason because those lookups aren't in the attribute lookup path so this only speeds up that first attribute lookup per module so didn't think it was worth it - I feel like the indirection of a new variable + the cost of creating the new variable isn't worth the cost of changing two global lookups and attribute accesses into two nonlocal lookups.

But I'm also not worried about it :smile: was just curious.